### PR TITLE
remove .lang from url and documents #100

### PR DESCRIPTION
**Pull request by dseeker**
maybe this is too broad and needs more precise definition, but it's working for me atm

### CODE REVIEW

File: lib/jekyll/polyglot/patches/jekyll/site.rb [hunk 1]
```diff
@@ -112,12 +113,28 @@ def process_documents(docs)
...
+        regex += "\\.#{lang}|"
+      end
+      regex.chomp! '|'
+      unless doc.url.nil?
+        doc.url.gsub!(%r{#{regex}}, "")
+      end
+      unless doc.output.nil?
+        doc.output.gsub!(%r{#{regex}}, "")
+      end
+    end
```
Comment (untra) [inline lib/jekyll/polyglot/patches/jekyll/site.rb]: I would like just a few example tests written for this function `cleanLang` (shouldn't be too hard, just stick 'em in [here](https://github.com/untra/polyglot/blob/master/spec/tests/lib/jekyll/polyglot/patches/jekyll/site_spec.rb) on the branch) , and I'll take a look at why ruby head is failing on travis.

One suggestion for writing ruby functions: the last run statement is the return of the function. So if either `.gsub!` call runs, cleanLang is returning a string (see [gsub! ruby-doc](https://ruby-doc.org/core-2.4.2/String.html#method-i-gsub-21)) , but if both doc.url and doc.output are nil, then the last valid statement `regex.chomp! '|'` is returned. This means cleanLang is returning a regex, a different type.

to assist with writing the test, please just return the mutated `doc` at the end of the function 👍 
```suggestion
      doc
    end
```

**Comment by untra** [comment]
Heya @dseeker , heya @funkytraveler 👋 

love the PR 😍 I am always looking for community support around this project. Please go ahead and add a few tests around `cleanLang`. I'll take a look at the travis status later tonight when I have some personal time and figure out why builds are failing on `ruby-head`, and then we can get this merged in tomorrow or later in the week. A new gem release will need to be cut and published before the changes are fully in; that'll be a separate PR after this gets in.

Cheers!

**Comment by untra** [comment]
@dseeker @funkytraveler make another commit to this PR with my suggestions above and I'll help get things over the line. #103 so the travis light should turn green now.

**Comment by untra** [comment]
@dseeker if you still want to get this in, feel free to open a new PR and get a few tests in. Otherwise I'm going to close this PR for now.

**Comment by dseeker** [comment]
@untra feel free to use this code, I won't be making further changes

**Pull request closed without merging**